### PR TITLE
Feature/unauthorized user can view the contacts relationships

### PR DESCRIPTION
**Pull request by mdeleon0**
[Patrick comment]
Included Relationship access check since we should be respecting that permission as well, variants of error messaging now appear as: 

Mix of fields are inaccessible: 
screen shot 2018-12-18 at 5 32 14 pm

Both Contacts and Relationship being unavailable:
screen shot 2018-12-18 at 10 40 38 am

Only relationships unavailable:
screen shot 2018-12-18 at 10 41 27 am
[/Patrick]

# Critical Changes

# Changes

# Issues Closed

# New Metadata

# Deleted Metadata

### CODE REVIEW

**Comment by lmeerkatz** [comment]
**lurch:attach W-024269

File: src/classes/REL_RelationshipsViewer_CTRL.cls [hunk 1]
```diff
@@ -34,92 +34,145 @@
...
+                                toLabel(npe4__Type__c),
+                                npe4__Status__c,
+                                npe4__RelatedContact__r.FirstName,
+                                npe4__RelatedContact__r.id,
+                                npe4__RelatedContact__r.Account.Name,
+                                npe4__RelatedContact__r.Title,
+                                npe4__ReciprocalRelationship__r.id
+                        FROM    npe4__Relationship__c
+                        WHERE   npe4__Contact__c = :contactId AND
+                                id NOT IN :renderedRelationships AND
```
Comment (lmeerkatz) [inline src/classes/REL_RelationshipsViewer_CTRL.cls]: id -> Id (also on next line)

Comment (lmeerkatz) [reply src/classes/REL_RelationshipsViewer_CTRL.cls]: and two more above this

File: src/classes/REL_RelationshipsViewer_CTRL.cls [hunk 2]
```diff
@@ -34,92 +34,145 @@
...
+                                npe4__ReciprocalRelationship__r.id NOT IN :renderedRelationships];
+
+        for (npe4__Relationship__c r : relationships) {
```
Comment (lmeerkatz) [inline src/classes/REL_RelationshipsViewer_CTRL.cls]: Avoid single character variable names. I know the style guide allows for them in small for loops, but I'd rather see them avoid when the letter has some meaning.

**Comment by LurchTheButler** [comment]
Tracking <a href="https://foundation.my.salesforce.com/a2x8000000AVvPgAAL" target="_blank">W-024269</a>

File: src/classes/REL_RelationshipsViewer_CTRL.cls (hunk 1 above)
Comment (mdeleon0) [reply src/classes/REL_RelationshipsViewer_CTRL.cls]: I missed those, fining them now.

File: src/classes/REL_RelationshipsViewer_CTRL.cls (hunk 2 above)
Comment (mdeleon0) [reply src/classes/REL_RelationshipsViewer_CTRL.cls]: Agreed

File: src/classes/REL_RelationshipsViewer_CTRL.cls [hunk 3]
```diff
@@ -34,92 +34,145 @@
...
+        showNoAccess();
+    }
+
+    /*******************************************************************************************************
+    * @description Sets the flag and error message if the current user does not have permission on he Contacts object.
+    * @return Pagereference to the current page.
+    */
+    private Pagereference showNoAccess() {
+
+        if (!Contact.sObjectType.getDescribe().isAccessible()) {
```
Comment (ptewson-sfdo) [inline src/classes/REL_RelationshipsViewer_CTRL.cls]: Should we be checking for relationship access as well?

File: src/pages/REL_RelationshipsViewer.page [hunk 4]
```diff
@@ -262,85 +262,73 @@
...
-                                if (result[i].FirstName === "undefined" || result[i].FirstName == null){
-                                    thisConFirstName = '';
-                                }
-                                else{ 
-                                    thisConFirstName = result[i].FirstName; 
-                                }
+                var showPermissionError = '{!showPermissionError}';
+				console.log('showPermissionError',showPermissionError);
+                if (showPermissionError == "false") {
+                    console.log('The initial call to the controller');
```
Comment (ptewson-sfdo) [inline src/pages/REL_RelationshipsViewer.page]: Remove console.log statements

File: src/pages/REL_RelationshipsViewer.page [hunk 5]
```diff
@@ -262,85 +262,73 @@
...
-                                }
-                                var thisConFirstName = '';
-                                if (result[i].FirstName === "undefined" || result[i].FirstName == null){
-                                    thisConFirstName = '';
-                                }
-                                else{ 
-                                    thisConFirstName = result[i].FirstName; 
-                                }
+                var showPermissionError = '{!showPermissionError}';
+				console.log('showPermissionError',showPermissionError);
```
Comment (ptewson-sfdo) [inline src/pages/REL_RelationshipsViewer.page]: Remove console.log statements

**Pull request merged**